protocol/ntp/packet.go

```diff
@@ -151,19 +151,14 @@ func ReadNTPPacket(conn *net.UDPConn) (ntp *Packet, remAddr net.Addr, err error)
 
 // ReadPacketWithKernelTimestamp reads HW/kernel timestamp from incoming packet
 func ReadPacketWithKernelTimestamp(conn *net.UDPConn) (ntp *Packet, hwRxTime time.Time, remAddr net.Addr, err error) {
-	// Get socket fd
-	connfd, err := connFd(conn)
-	if err != nil {
-		return nil, time.Time{}, nil, err
-	}
 	buf := make([]byte, PacketSizeBytes)
 	oob := make([]byte, ControlHeaderSizeBytes)
 
 	// Receive message + control struct from the socket
 	// https://linux.die.net/man/2/recvmsg
 	// This is a low-level way of getting the message (NTP packet content)
 	// Additionally we receive control headers, one of which is hwtimestamp
```

Comment:
Is this still accurate?

Reply:
Yes, I think so. ReadMsgUDP is just a wrapper around that syscall.

https://golang.org/pkg/net/#UDPConn.ReadMsgUDP